plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/MachineProviderImpl.java

```diff
@@ -591,6 +617,39 @@ private void addSystemWideContainerSettings(String workspaceId,
         composeService.getNetworks().addAll(additionalNetworks);
     }
 
+    // We can detect certain situation when container exited right after start.
+    // We can detect
+    //  - when no command/entrypoint is set
+    //  - when most common shell interpreters are used and require additional arguments
+    //  - when most common shell interpreters are used and they require interactive mode which we don't support
+    // When we identify such situation we change CMD/entrypoint in such a way that it runs "tail -f /dev/null".
+    // This command does nothing and lasts until workspace is stopped.
+    // Images such as "ubuntu" or "openjdk" fits this situation.
+    protected void setNonExitingContainerCommandIfNeeded(ContainerConfig containerConfig) throws IOException {
+        ImageConfig imageConfig = docker.inspectImage(containerConfig.getImage()).getConfig();
+        List<String> cmd = imageConfig.getCmd() == null ?
+                           null : Arrays.asList(imageConfig.getCmd());
+        List<String> entrypoint = imageConfig.getEntrypoint() == null ?
+                                  null : Arrays.asList(imageConfig.getEntrypoint());
+
+        if ((entrypoint == null || badEntrypoints.contains(entrypoint)) && (cmd == null || badCMDs.contains(cmd))) {
+            containerConfig.setCmd("tail", "-f", "/dev/null");
+            containerConfig.setEntrypoint((String[])null);
+        }
+    }
+
+    // Inspect container right after start to check if it is running,
+    // otherwise throw error that command should not exit right after container start
+    protected void checkContainerIsRunning(String container) throws IOException, ServerException {
+        ContainerInfo containerInfo = docker.inspectContainer(container);
+        if ("exited".equals(containerInfo.getState().getStatus())) {
+            throw new ServerException("We detected that machine container exited right after start. " +
```

Comment:
Please change message to: 
>We detected that a machine exited unexpectedly. This may be caused by a container in interactive mode or a container that requires additional arguments to start. Please check the container recipe.